web/modules/custom/solution/src/EventSubscriber/WorkflowTransitionEventSubscriber.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+namespace Drupal\solution\EventSubscriber;
+
+use Drupal\message\Entity\Message;
+use Drupal\message\Entity\MessageTemplate;
+use Drupal\message_notify\MessageNotifier;
+use Drupal\og\Entity\OgMembership;
+use Drupal\state_machine\Event\StateChangeEvent;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+/**
+ * Event subscribers for Organic Groups.
+ */
+class WorkflowTransitionEventSubscriber implements EventSubscriberInterface {
+
+
+  /**
+   * The message notifier service.
+   *
+   * @var \Drupal\message_notify\MessageNotifier
+   */
+  protected $messageNotifier;
+
+  /**
+   * Constructs an WorkflowTransitionEventSubscriber object.
+   */
+  public function __construct(MessageNotifier $messageNotifier) {
+    $this->messageNotifier = $messageNotifier;
+  }
+
+  /**
+   * On workflow transition.
+   *
+   * @param \Drupal\state_machine\Event\StateChangeEvent $event
+   *   The state change event.
+   *
+   * @throws \Drupal\message_notify\Exception\MessageNotifyException
+   */
+  public function onChangeToValidated(StateChangeEvent $event) {
+    $entity = $event->getEntity();
+    $bundle = $entity->bundle();
+    if ($bundle != 'solution') {
+      return;
+    }
+    $message_template = MessageTemplate::load('workflow_transition');
+
+    $storage = \Drupal::entityManager()->getStorage('og_membership');
+
+    // Sent a message to all solution administrators.
+    $membership_query = $storage->getQuery();
+    $membership_query
+      ->condition('state', 'active')
+      ->condition('entity_id', $entity->id())
+      ->condition('roles', 'rdf_entity-solution-administrator');
```

Comment:
Hm. I'm not very happy with checking for a specific **role** in code. By doing this we lock ourself to that role. Instead, checking by a specific permission is a better approach. eg. "receive solution notification". Then, when we want to extend the target of notifications we only add grant that new role with this permission and we're done.